### PR TITLE
[W5.3b][W13-A2] Aadyaa Maddi

### PR DESCRIPTION
**Pull request by amad-person**
LO-Inheritance: 
Extracted commonalities from Phone, Email and Address classes into a parent class Contact.

### CODE REVIEW

File: src/seedu/addressbook/data/person/Contact.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Represents a Person's common details in a contact field in the address book.
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:4]: `Represents a person's contact information` would suffice here.

File: src/seedu/addressbook/data/person/Contact.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ */
+public class Contact {
+
+    private String value;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:9]: :+1:

File: src/seedu/addressbook/data/person/Contact.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        this.isPrivate = isPrivate;
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public int hashCode() {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:23]: In general, `equals()` and `hashCode()` methods should be overridden together. Find out why!

**Review by okkhoy — COMMENTED** [review]
You can try to move the validation logic as well to the superclass.

**Comment by okkhoy** [comment]
@amad-person 
Some comments added. Please ack & close the PR after reading comments.

**Comment by amad-person** [comment]
Noted, thanks!

**Pull request closed without merging**